dashboards-observability/public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -751,9 +750,34 @@ export const Explorer = ({
       setTriggerAvailability(false);
     }
   };
-
-  const changeIsValidConfigOptionState = (isValidConfig: boolean) =>
-    setIsValidDataConfigOptionSelected(isValidConfig);
+  const isValidValueOptionConfigSelected = useMemo(() => {
+    const { series = [], dimensions = [], span = {} } = visualizations.data.userConfigs?.dataConfig;
+    const { TreeMap, Gauge, HeatMap, Stats } = VIS_CHART_TYPES;
+    const isValidValueOptionsXYAxes =
```

Comment:
I would normally split up this logic for readability/comprehension purposes.

The end-logic I would want to read would be something like : 
```
const isValidValueOptionsYAxes = VIZ_CONTAIN_XY_AXIS.includes(curVisId as VIS_CHART_TYPES) &&
          (chartDataPresent(series, dimensions) || spanIsPresent(span) )
 ```
 Just an example - comprehension for an engineer who is new or has not read the code in several monthts
 is the higher priority